dss-apps/dss-datamodel-center/dss-datamodel-indicator-management/src/main/java/com/webank/wedatasphere/dss/datamodel/indicator/service/impl/IndicatorServiceImpl.java

```diff
@@ -0,0 +1,539 @@
+                ||datamodelReferencService.indicatorReferenceCount(org.getFieldIdentifier())>0
+                ||datamodelReferencService.indicatorReferenceCount(vo.getFieldIdentifier())>0){
+            LOGGER.error("errorCode : {}, indicator name can not  referenced", ErrorCode.INDICATOR_UPDATE_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_UPDATE_ERROR.getCode(), "indicator name can not referenced");
+        }
+
+
+        DssDatamodelIndicatorContent orgContent = indicatorContentService.queryByIndicateId(id);
+
+
+        //当更新指标名称时,存在其他指标名称同名或者当前指标名称已经存在版本信息，则不允许修改指标名称
+        if (!StringUtils.equals(vo.getName(), org.getName())) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelIndicator>lambdaQuery().eq(DssDatamodelIndicator::getName, vo.getName()));
+            String lastVersion = indicatorVersionService.findLastVersion(org.getName());
+            if (repeat > 0 || StringUtils.isNotBlank(lastVersion)) {
+                LOGGER.error("errorCode : {}, indicator name can not repeat or has version", ErrorCode.INDICATOR_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_UPDATE_ERROR.getCode(), "indicator name can not repeat or has version");
+            }
+        }
+
+        String orgFieldIdentifier =  org.getFieldIdentifier();
+        //当更新指标标识时,存在其他指标名称同名或者当前指标名称已经存在版本信息，则不允许修改指标名称
+        if (!StringUtils.equals(vo.getFieldIdentifier(), orgFieldIdentifier)) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelIndicator>lambdaQuery().eq(DssDatamodelIndicator::getFieldIdentifier, vo.getFieldIdentifier()));
+            if (repeat > 0) {
+                LOGGER.error("errorCode : {}, indicator field identifier can not repeat or has version", ErrorCode.INDICATOR_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_UPDATE_ERROR.getCode(), "indicator field identifier can not repeat or has version");
+            }
+        }
+
+
+        Long orgId = org.getId();
+        String version = org.getVersion();
+
+        //更新原有指标
+        DssDatamodelIndicator updateOne = modelMapper.map(vo, DssDatamodelIndicator.class);
+        updateOne.setUpdateTime(new Date());
+        int result = getBaseMapper().update(updateOne, Wrappers.<DssDatamodelIndicator>lambdaUpdate().eq(DssDatamodelIndicator::getId, id));
+        if (result != 1) {
+            LOGGER.error("errorCode : {}, update indicator error not exists", ErrorCode.INDICATOR_UPDATE_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_UPDATE_ERROR.getCode(), "update indicator error not exists");
+        }
+
+        indicatorContentService.updateIndicatorContent(orgId, version, vo.getContent());
+
+
+
+        //同步atlas
+        UpdateModelTypeResult updateModelTypeResult = assertsSyncService.syncUpdateModel(
```

Comment:
publisher.publishEvent(new UpdateModelEvent(this
                ,DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
                ,vo.getFieldIdentifier()
                ,orgFieldIdentifier
                ,ClassificationConstant.INDICATOR));